patterns/solid/src/main/java/com/baeldung/l/advanced/ReadOnlyFileSystem.java

```diff
@@ -0,0 +1,16 @@
+package com.baeldung.l.advanced;
+
+import java.io.File;
+import java.io.IOException;
+
+public class ReadOnlyFileSystem implements FileSystem {
+    public File[] listFiles(String path) {
+        // code to list files
+        return new File[0];
+    }
+
+    public void deleteFile(String path) throws IOException {
+        // Do nothing.
```

Comment:
Do nothing? or throw exception as per other examples?

Reply:
As you picked up during the article review, this was deliberate to illustrate a code smell. 
"Doing Nothing" with this comment 😉